File: src/main/java/ninekothecat/catconomy/interfaces/IPermissionGuard.java
package ninekothecat.catconomy.interfaces;

/**
 * The interface Permission guard.
 */
public interface IPermissionGuard {
    /**
     * checks if the transaction is permitted.
     *
     * @param transaction the transaction
     * @return the boolean
     */
    boolean isPermitted(ITransaction transaction);
}

File: src/main/kotlin/ninekothecat/catconomy/interfaces/IPermissionGuard.kt
package ninekothecat.catconomy.interfaces

import ninekothecat.catplugincore.money.interfaces.ITransaction

/**
 * The interface Permission guard.
 */
interface IPermissionGuard {
    /**
     * checks if the transaction is permitted.
     *
     * @param transaction the transaction
     * @return the boolean
     */
    fun isPermitted(transaction: ITransaction): Boolean
}